### PR TITLE
'Starter Project': port planet fragment shader to impeller tests

### PR DESCRIPTION
**Pull request by jtmcdole**
This is a modified planet shader from the SimonDev's "GLSL Shaders from Scratch" course on teachables. With permission from Simon:

> 100% use however you want, with or without credit

I choose "with credit".

https://github.com/flutter/engine/assets/1924313/47da62b5-6b6d-47c2-ac9f-f3a33f177a86

### CODE REVIEW

File: impeller/fixtures/planet.frag [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+float sdfCircle(vec2 p, float r) {
+  return length(p) - r;
+}
+
+float map(vec3 pos) {
+  return fbm6(pos, 0.5, 2.0, 4.0);
+}
+
+vec3 calcNormal(vec3 pos, vec3 n) {
+  // if you sample the noise field along each axis, a small amount of distnace
```
Comment (jonahwilliams) [inline impeller/fixtures/planet.frag:310]:
```suggestion
  // if you sample the noise field along each axis, a small amount of distance
```

File: impeller/fixtures/planet.frag [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+  vUvs.y = 1.0 - vUvs.y;  // flip flutter's upside down.
+  vec2 pixel_coords = (vUvs - 0.5) * frag_info.resolution;
+
+  vec3 color = vec3(0.0);
+  color = GenerateStars(pixel_coords);
+  color = DrawPlanet(pixel_coords, color, frag_info.planet_size,
+                     frag_info.planet_size * 0.1, frag_info.speed);
+
+  frag_color = vec4(pow(color, vec3(1.0 / 2.2)), 1.0);
+}
```
Comment (bdero) [inline impeller/fixtures/planet.frag:486]: Needs newline

**Review by bdero — APPROVED** [review]
Cool, LGTM!

File: impeller/fixtures/planet.frag (hunk 2 above)
Comment (jtmcdole) [reply impeller/fixtures/planet.frag:486]: Is this something `et format` needs to handle better?

Comment (bdero) [reply impeller/fixtures/planet.frag]: Yeah not sure why clang-format isn't doing this anymore. Filed a quick issue: https://github.com/flutter/flutter/issues/150148

**Pull request merged**